lib/pure/terminal.nim

```diff
@@ -677,3 +758,21 @@ when not defined(testing) and isMainModule:
   stdout.setForeGroundColor(fgBlue)
   stdout.writeLine("ordinary text")
   stdout.resetAttributes()
+
+when not defined(windows):
+  styleCache = newTable[int, string]()
+
+fgSetColor = true
+colorsFGCache = newTable[Color, string]()
```

Comment:
This only runs for the main thread!

Reply:
Sorry I really do not understand what is wrong.

Reply:
He means that the ``colorsFGCache`` will only be initialised in the main thread. If you try using it from another thread it will be ``nil`` and your code will likely fail.

Reply:
Thanks!
I looked at the code in the /tests/parallel/tsendtwice.nim.
It works there. Or not?